api/src/main/java/io/minio/NotificationListener.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.minio;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.minio.notification.NotificationInfo;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Scanner;
+
+/**
+ * NotificationListener is used for listening notification.
+ */
+public class NotificationListener  {
+
+  Scanner sc = null;
+  String notificationString = "";
+  NotificationInfo ni = null;
+  Reader inputReader;
+  boolean isClosed = false;
+  ObjectMapper mapper = new ObjectMapper();
+
+  NotificationListener( Reader reader) {
+    this.inputReader = reader;
+
+    this.sc = new Scanner(inputReader);
+    this.sc.useDelimiter("\n");
+  }
+  /**
+   * Return a closeable iterator.
+   */
+
+  public  CloseableIterator<Result<NotificationInfo>> getMyCloseableIterator() {
+    return new CloseableIterator<Result<NotificationInfo>>() {
+
+      @Override
+      public boolean hasNext() {
+        if (isClosed) {
+          return false;
+        }
+        if (!NotificationListener.this.sc.hasNext()) {
+          try {
+            close();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+          return false;
+        }
+        return true;
+      }
+
+      @Override
+      public Result<NotificationInfo> next() {
+        try {
+          while (sc.hasNext()) {
```

Comment:
this `while` should be in `hasNext()`

Reply:
When i tried to have the while in hasNext() as shown in the code below 
```
@Override
      public boolean hasNext()  {
        if (isClosed) {
          return false;
        }
        if (!scanner.hasNext()) {
          try {
            close();
          } catch (IOException e) {
            exception = e;
          }
          return false;
        }

        while (scanner.hasNext()) {
          notificationString = scanner.next().trim();
          if (notificationString.equals("")) {
            continue;
          } else {
            break;
          }

        }
        return true;
      }

      @Override
      public Result<NotificationInfo> next() {
        try {
          notificationInfo = mapper.readValue(notificationString, NotificationInfo.class);
        } catch (JsonParseException e) {
          exception = e;
        } catch (  JsonMappingException mappingEx) {
          exception = mappingEx;
        } catch ( IOException ioexception) {
          exception = ioexception;
        }
        if ( exception != null ) {
          return new Result<>(null, exception);
        }
        return new Result<>(notificationInfo, null);
      }
```
Got a build error as shown 
```
io.minio.MinioClient$8.next() can't throw NoSuchElementException
 ```

Thus while is not a part of `hasNext`